scripts/zenhub_close.py

```diff
@@ -0,0 +1,137 @@
+"""
+This script closes issues in the Ready for Prod pipeline that are not blocked by open issues or PRs.
+"""
+import json
+import logging
+import os
+import sys
+from typing import Iterable, List, Tuple
+
+import requests
+
+logging.basicConfig(level=logging.INFO, format="%(message)s", handlers=[logging.StreamHandler(sys.stdout)])
+logger = logging.getLogger(__name__)
+
+pipeline_id = "Z2lkOi8vcmFwdG9yL1BpcGVsaW5lLzIyMDg1MDE"  # "Ready for Prod" pipeline in "single-cell" workspace
+repo_id = "Z2lkOi8vcmFwdG9yL1JlcG9zaXRvcnkvNTM5NzQwOTg"  # "single-cell-data-portal" repo
+
+
+def get_issues(pipeline_id: str, repo_id: str) -> List[dict]:
+    query_list_issues = """query {
+    searchIssuesByPipeline(
+        pipelineId: "$PIPELINE_ID",
+        filters: {
+          repositoryIds: "$REPO_ID"
+        }
+    ) {
+      nodes {
+          id
+          number
+          title
+          blockingIssues{
+            nodes{
+              id
+              number
+              title 
+              state
+            }
+          }
+          connectedPrs{
+            nodes{
+              id
+              number
+              title
+              state
+            }
+          }
+      }
+    }
+    }""".replace(
+        "$PIPELINE_ID", pipeline_id
+    ).replace(
+        "$REPO_ID", repo_id
+    )
+    resp = requests.post(endpoint, json={"query": query_list_issues}, headers=headers)
+    resp.raise_for_status()
+    return resp.json()["data"]["searchIssuesByPipeline"]["nodes"]
+
+
+def close_issues(issue_ids: Iterable[str]) -> dict:
+    mutate_close_isses = """mutation closeIssues {
+      closeIssues(input: {
+          issueIds: $CLOSING_ISSUES
+      }) {
+          successCount
+      }
+    }""".replace(
+        "$CLOSING_ISSUES", json.dumps(issue_ids)
+    )
+    resp = requests.post(endpoint, json={"query": mutate_close_isses}, headers=headers)
+    resp.raise_for_status()
+    return resp.json()
+
+
+def format_issue(issue: dict) -> str:
+    return f"#{issue['number']} ({issue['title']})"
+
+
+def filter_issues(issues: List[dict]) -> Tuple[List[Tuple[str, str]], List[dict]]:
```

Comment:
I'm a bit iffy on this function name since it is generic and implies that it might take filter parameters but not a huge deal I guess